src/kudu/consensus/quorum_util.cc

```diff
@@ -922,10 +945,61 @@ void AdjustVoterDistributionWithCurrentVoters(
   }
 }
 
+void GetVoterDistributionForQuorumId(
+    const RaftConfigPB& config,
+    std::map<std::string, int>* quorum_id_vd
+) {
+  if (IsUseQuorumId(config.commit_rule())) {
+    // Step 1: Using the default quorum size
+    for (const auto peer : config.peers()) {
+      if (peer.has_member_type() && peer.member_type() == RaftPeerPB::VOTER) {
+        CHECK(peer.has_attrs() && peer.attrs().has_quorum_id());
+        InsertIfNotPresent(quorum_id_vd, peer.attrs().quorum_id(), FLAGS_default_quorum_size);
+      }
+    }
+
+    // Step 2: Adjust vd based on current vd overrides
+    for (const auto original_vd_entry : config.voter_distribution()) {
+      auto it = quorum_id_vd->find(original_vd_entry.first);
```

Comment:
we need a comment here which says that even if there is a SKEW between voter distribution and quorum id usage during rollouts, this loop should not alter the voter count map, because quorum id will have a proper prefix.